### DISCUSSION TITLE
Prevent removal of user attributes mapped from claims?

### DISCUSSION THREAD

**Discussion by sys-jdi** [keycloak/keycloak#12456] [Q&A]
Hi Keycloak community!

In my team we've setup Keycloak to allow login using an external OIDC identity provider. The id token received from the identity provider contains different claims based on the scopes set when initiating the login flow.

What we would like to achieve is mapping one of these claims to the logged in user as an attribute when the claim is present in the received token. This is simple enough with a claims mapper configured in Keycloak for the given IDP. However the user attribute is removed again if a later login does not result in an ID token containing the claim.

Is there any way to prevent the removal of user attributes in this way? 

Implementing a custom claims mapper could be one way however we would prefer if it was possible with configuration somehow.

**Comment by Torrien** [comment]
Hi @sys-jdi did you find a solution to your need? We are considering a similar behavior and would like to know if you found a built-in solution?

**Reply by sys-jdi** [reply]
Hi Torrien. 
Unfortunately no, we didn't find a solution to this so we implemented our system in a different way.
Perhaps later versions of Keycloak do support it as we started this work on version 16 or 17 I believe.
